hawc/apps/common/filterset.py

```diff
@@ -201,20 +195,20 @@ def perms(self):
     @property
     def form(self):
         if not hasattr(self, "_form"):
-            self._form = self.create_form()
+            form = self.get_form_class()
+            if self.is_bound:
+                _form = form(self.data, prefix=self.form_prefix, **self.form_kwargs)
+            else:
+                _form = form(prefix=self.form_prefix, **self.form_kwargs)
+            if _form.dynamic_fields:
+                for field in list(_form.fields.keys()):
+                    if field not in _form.dynamic_fields and field != "is_expanded":
+                        _form.fields.pop(field)
+            self._form = self.customize_form(_form)
         return self._form
 
-    def create_form(self) -> forms.Form:
-        """Create the form used for the filterset.
-
-        Returns:
-            forms.Form: a django.Form instance
-        """
-        Form = self.get_form_class()
-        if self.is_bound:
-            form = Form(self.data, prefix=self.form_prefix, grid_layout=self._meta.grid_layout)
-        else:
-            form = Form(prefix=self.form_prefix, grid_layout=self._meta.grid_layout)
+    def customize_form(self, form):
```

Comment:
maybe we dont need this? it may be better to add a get_form method in the view, and then if you want to customize you can do there?